pkg/heartbeat/heartbeat.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package heartbeat
```

Comment:
Yes, it's tightly coupled to components. I had a more generic http heartbeat, but it turned out to be unnecessary, given the use of the KubeClient. I could probably make a more generic solution with two implementations, but I try not to get into premature abstraction without a use case.

I don't know anything about the podmaster. It'd be nice if there were a readme...